src/components/PasswordDialog.vue

```diff
@@ -21,54 +21,52 @@
 -->
 
 <template>
-	<NcModal :id="dialogId"
-		class="dialog"
-		size="small"
+	<NcDialog :id="dialogId"
+		:name="dialogName"
 		:container="null"
-		@close="close">
-		<div class="dialog__container">
-			<h2 class="dialog__title">
-				{{ titleText }}
-			</h2>
-			<p>{{ subtitleText }}</p>
-
-			<NcPasswordField ref="field"
-				:value.sync="password"
-				:label="passwordLabelText"
-				:helper-text="showError ? errorText : ''"
-				:error="showError"
-				required
-				@keydown.enter="confirm" />
-
+		contentClasses="vue-password-confirmation"
+		@update:open="close">
+		<template #actions>
 			<NcButton type="primary"
-				class="dialog__button"
 				:disabled="!password"
-				:aria-label="confirmText"
 				@click="confirm">
 				{{ confirmText }}
 			</NcButton>
-		</div>
-	</NcModal>
+		</template>
+		<!-- Dialog content -->
+		<p>{{ subtitleText }}</p>
+		<NcPasswordField ref="field"
+			:value.sync="password"
+			:label="passwordLabelText"
+			:helper-text="showError ? errorText : ''"
+			:error="showError"
+			required
+			@keydown.enter="confirm" />
```

Comment:
I think it makes sense semantically to move this into a form and confirm on submit

And also to automatically highlight the entered password it's invalid

Reply:
I believe that this is already covered via the `:error="showError"` attribute. 

Semantically, I would agree that a form does fit better, but seeing as this only has one relevant field to submit, this works just fine IMO 👍

Reply:
> I believe that this is already covered via the `:error="showError"` attribute.

😅 meant that the password text should be selected like in this screenshot
![image](https://github.com/nextcloud-libraries/nextcloud-password-confirmation/assets/24800714/7806e8e9-cf83-4d2b-92c3-385414dafdfb)
so the user can hit backspace to erase it and retype again

Reply:
Ahhh, my mistake. I agree with this being implemented then :)

Reply:
implemented